src/rust/lib/graph-editor/src/component/visualization/js.rs

```diff
@@ -0,0 +1,125 @@
+//! This module contains functionality to create a `Class` object from a JS source strings.
+
+use crate::prelude::*;
+
+use crate::component::visualization::JsVisualizationError;
+use crate::component::visualization::JsRenderer;
+use crate::component::visualization::JsResult;
+use crate::component::visualization::InstantiationResult;
+use crate::component::visualization::Class;
+use crate::component::visualization::Visualization;
+use crate::component::visualization::ClassAttributes;
+use crate::component::visualization::EnsoType;
+
+use ensogl::display::Scene;
+use ensogl::system::web::JsValue;
+use js_sys;
+
+
+
+// ===================================
+// === Visualization Class Wrapper ===
+// ===================================
+
+/// Internal wrapper for the a JS class that implements our visualization specification. Provides
+/// convenience functions for accessing JS methods and attributes.
+#[derive(Clone,Debug)]
+#[allow(missing_docs)]
+struct VisualizationClassWrapper {
+    class: JsValue,
+}
+
+impl VisualizationClassWrapper {
+    fn instantiate_class(source:&str) -> VisualizationClassWrapper {
+        let context     = JsValue::NULL;
+        let constructor = js_sys::Function::new_no_args(source);
+        let class       = constructor.call0(&context).unwrap();
+        VisualizationClassWrapper{class}
+    }
+
+    fn attributes(&self) -> JsResult<ClassAttributes> {
+        let input_types = self.input_types().unwrap_or_default();
+        let name        = self.name()?;
+        Ok(ClassAttributes{name,input_types})
+    }
+
+    fn constructor(&self) -> js_sys::Function {
+        js_sys::Reflect::get(&self.prototype(),&"constructor".into()).unwrap().into()
```

Comment:
JS world is very very strange. You can alter everything there. And we never want that user provided visualization crashes the application, thus you should never do `unwrap` when accessing properties in user-provided code. We can use `unwrap` when we own the behavior, like for example getting some webgl context, but on user-provided code we need to be very, very careful.

Reply:
I fixed all the error handling here. There should be no `unwrap`s left.